### PR TITLE
Add metadata to build SPL packages for Ubuntu

### PR DESCRIPTION
**Pull request by pzakha**
## Commits

**Add Ubuntu packaging metadata to the repository**

Created by running uupdate with spl-linux 0.7.5-1ubuntu2.
Removed all patches and added Delphix template version in changelog.

## Description
This is the SPL part for the ZFS packaging. Main PR is: https://github.com/delphix/zfs/pull/5.
Note that no additional changes were required from the original Ubuntu package besides
removing patches and updating the changelog to reflect Delphix versioning.

## Testing
This was tested along with the ZFS PR mentioned in the description.

### CODE REVIEW

File: config/spl-build.m4 [hunk 1]
```diff
@@ -228,7 +225,9 @@ AC_DEFUN([SPL_AC_RPM], [
 		AC_MSG_RESULT([$HAVE_RPMBUILD])
 	])
 
-	RPM_DEFINE_COMMON='--define "$(DEBUG_SPL) 1" --define "$(DEBUG_KMEM) 1" --define "$(DEBUG_KMEM_TRACKING) 1"'
+	RPM_DEFINE_COMMON='--define "$(DEBUG_SPL) 1"'
+	RPM_DEFINE_COMMON+=' --define "$(DEBUG_KMEM) 1"'
+	RPM_DEFINE_COMMON+=' --define "$(DEBUG_KMEM_TRACKING) 1"'
```
Comment (ahrens) [inline config/spl-build.m4:230]: Is this a stylistic / code cleanup change?  If so, we should make it upstream instead.

File: configure.ac [hunk 2]
```diff
@@ -46,6 +46,9 @@ AC_PROG_LIBTOOL
 SPL_AC_LICENSE
 SPL_AC_PACKAGE
 SPL_AC_CONFIG
+SPL_AC_DEBUG
+SPL_AC_DEBUG_KMEM
+SPL_AC_DEBUG_KMEM_TRACKING
```
Comment (ahrens) [inline configure.ac:51]: Can you explain why we we move these lines from spl-build.m4 to here?  I don't know what they do.

**Review by ahrens — COMMENTED** [review]
I think there's an opportunity for a debian/README file that would explain why we imported these files here, where they came from, when we would pull changes from debian, and how to do so.

(Though the main README probably belongs in the zfs repo, and a README here in spl could point to that one.)

File: config/spl-build.m4 (hunk 1 above)
Comment (pzakha) [reply config/spl-build.m4:230]: Hmm I believe those changes are from the upstream commit https://github.com/delphix/spl/pull/3/commits/84a80d5f2d14fdfd16d440eef2e3d6e756a1165c which ended up in this PR when I opened it. Still getting familiar with the github workflow...

File: configure.ac (hunk 2 above)
Comment (pzakha) [reply configure.ac:51]: Same as above.

**Comment by ahrens** [comment]
You'll need to make these changes to the zfs repo instead, now that spl has been moved there.

**Pull request closed without merging**